superset/daos/chart.py

```diff
@@ -39,19 +37,13 @@ class ChartDAO(BaseDAO[Slice]):
     base_filter = ChartFilter
 
     @classmethod
-    def delete(cls, items: Slice | list[Slice], commit: bool = True) -> None:
+    def delete(cls, items: Slice | list[Slice]) -> None:
         item_ids = [item.id for item in get_iterable(items)]
-        # bulk delete, first delete related data
-        # bulk delete itself
-        try:
+
+        with db.session.begin_nested():
             db.session.query(Slice).filter(Slice.id.in_(item_ids)).delete(
                 synchronize_session="fetch"
             )
-            if commit:
-                db.session.commit()
-        except SQLAlchemyError as ex:
-            db.session.rollback()
-            raise ex
```

Comment:
We're really inconsistent with our error handling. The `BaseDAO.delete` method wraps all `SQLAlchemyError` errors as `DAODeleteFailedError` whereas here they are left as is.